sqlglot/optimizer/simplify.py

```diff
@@ -557,9 +562,9 @@ def _simplify_binary(expression, a, b):
     elif _is_date_literal(a) and isinstance(b, exp.Interval):
         a, b = extract_date(a), extract_interval(b)
         if a and b:
-            if isinstance(expression, exp.Add):
+            if isinstance(expression, (exp.Add, exp.DateAdd, exp.DatetimeAdd)):
```

Comment:
can you add a test? dateadd -> date, datetimeadd -> datetime ?

Reply:
I _think_ I added the tests you want. Let me know if otherwise.